flashlight/memory/managers/CachingMemoryManager.cpp

```diff
@@ -126,7 +138,9 @@ void* CachingMemoryManager::alloc(
 
   CachingMemoryManager::Block* block = nullptr;
   auto it = pool.lower_bound(&searchKey);
-  if (it != pool.end()) {
+  // Recycle blocks if any found, and if small alloc or the block size is not
+  // too large:
+  if (it != pool.end() && (isSmallAlloc || (*it)->size_ < recyclingSizeLimit)) {
```

Comment:
Warning: now that recyclingSizeLimit is atomic, there could be a cost here to access/read it at each alloc.
https://stackoverflow.com/questions/2538070/atomic-operation-cost
Do you run inference/production with a high number of threads ?

Reply:
@WilliamTambellini — this is a good point - I actually think it's better that we don't make it atomic. I don't see a good use case for setting and getting it in multiple threads. Let's just add a comment to the doc block for the setters that says it isn't thread safe. Once that's done, we should be good to go!

Reply:
done